client/src/pages/queries/components/history-editor-tab.tsx

```diff
@@ -220,55 +236,152 @@ function Tab() {
+            <HorizontalPane
+              paneName="editor-horizontal-pane"
+              initialWidth={640}
+              maxConstraint={800}
+              minConstraint={320}
+              buffer={160}
+              render={({
+                paneWidth,
+                isFullScreen,
+                dragHandle,
+                toggleFullScreen,
+              }) => (
+                <>
                   <div
-                    className="w-full"
-                    style={{ height: containerBounds.height - paneHeight }}
+                    className="relative z-10 flex flex-col h-full border-r"
+                    style={{ width: paneWidth }}
                   >
                     <Editor
                       resourceId={selectedResourceId}
                       value={query}
                       setValue={setQuery}
                     />
+                    {dragHandle}
                   </div>
-                ) : null}
-                <div
-                  className="relative border-t"
-                  style={{ height: paneHeight }}
-                >
-                  {dragHandle}
-                  <div className="flex justify-end px-4 py-2">
-                    <button
-                      onClick={toggleFullScreen}
-                      className="focus:outline-none"
-                    >
-                      {isFullScreen ? <DownOutlined /> : <UpOutlined />}
-                    </button>
+                  <div
+                    className="h-full"
+                    style={{ width: containerBounds.width - paneWidth }}
+                  >
+                    <div className="flex justify-end px-4 py-2 space-x-4">
+                      <Tooltip
+                        title="Split Horizontally"
+                        placement="left"
+                        mouseEnterDelay={1}
+                      >
+                        <button
+                          className="focus:outline-none"
+                          onClick={() => {
+                            setSplitOrientation(SplitOrientation.HORIZONTAL)
+                          }}
+                        >
+                          <BorderHorizontalOutlined />
+                        </button>
+                      </Tooltip>
+                      <Tooltip
+                        title="Fullscreen"
+                        placement="left"
+                        mouseEnterDelay={1}
+                      >
+                        <button
+                          className="focus:outline-none"
+                          onClick={toggleFullScreen}
+                        >
+                          {isFullScreen ? <RightOutlined /> : <LeftOutlined />}
+                        </button>
+                      </Tooltip>
+                    </div>
+                    <div className="h-full px-4">
+                      <ResultsTable
+                        data={queryResult ?? undefined}
```

Comment:
We don't need to pass `?? undefined`. The `queryResult` should be either be results fetched or `undefined`.